drivers/platform/stm32/stm32_spi.h

```diff
@@ -66,6 +75,14 @@ struct stm32_spi_desc {
 	uint32_t input_clock;
 	/** Chip select gpio descriptor */
 	struct no_os_gpio_desc *chip_select;
+	/** DMA Descriptor */
+	struct no_os_dma_desc* dma_desc;
```

Comment:
that's good progress, i see fewer members in this structure but it can be improved further

this field shouldn't be here, actually the parent descriptor of this `stm32_spi_desc` is a `no_os_dma_desc` and you have direct access to that in the DMA API functions

Reply:
@buha  I am not sure I got the point right here. Do you mean to add the no_os_dma_desc to the no_os_spi_desc? We might need access to this DMA descriptor somewhere in the SPI descriptors so that it can be used in the async functions